helix-core/src/selection.rs

```diff
@@ -615,11 +615,6 @@ impl Selection {
 
     // returns true if self ⊇ other
     pub fn contains(&self, other: &Selection) -> bool {
-        // can't contain other if it is larger
-        if other.len() > self.len() {
```

Comment:
I am fairly certain this is the only change needed to get this working for the cases where there are big ranges in the first one that contain several smaller ranges in the second one. The algorithm is correct for these cases, this was just an incorrect attempt at a fast path.